src/main/scala/is/hail/expr/types/TContainer.scala

```diff
@@ -143,17 +143,17 @@ abstract class TContainer extends Type {
     clearMissingBits(region, aoff, length)
   }
 
-  def initialize(region: Code[Region], aoff: Code[Long], length: Code[Int], a: LocalRef[Int]): Code[Unit] = {
+  def initialize(region: Code[Region], aoff: Code[Long], length: Code[Int], a: Settable[Int]): Code[Unit] = {
     var c = region.storeInt(aoff, length)
     if (elementType.required)
       return c
     Code(
       c,
       a.store((length + 7) >>> 3),
-      Code.whileLoop(a > 0,
+      Code.whileLoop(a.load() > 0,
```

Comment:
And we need an implicit `toCode` for `Settable` as well? Not sure how the implicit resolution works with subtypes. Hopefully it shakes out OK?